spatialos/UnrealGDK
# [UNR-4765] remove entity id expiry

#### Description
Removed code that expired reserved entity IDs ranges after 3 minutes. This mean that reserved entity ID ranges now persist until completely used up.

#### Release note
[Internal] - Reserved entity IDs previously expired after 3 minutes. Reserved Entity IDs no longer expire, and persist until used.

#### Tests
I significantly lowered the three settings that determine when and how many new entity ids are reserved. I then checked the Example Project to make sure that there were no errors or warnings. I used these steps.
1. In the Unreal Editor navigate to **Project Settings > SpatialOS GDK for Unreal > Runtime Settings > Entity Pool > Initial Entity ID Reservation Count** and change this setting from the default `3000` to `60`.
1. Change the setting directly below, **Pool Refresh Threshold**  from the default `1000` to `20`.
1. Change the setting directly below, **Refresh Count**  from the default `2000` to `40`.
1. Play the Example Project in editor (generate schema first).
1. Run around, shoot, play the game for a couple of minutes.
1. Check the Output Log the confirm the absence of errors and warns.

I also checked this in the Handover Gym using similar steps.

What automated tests are included in this PR?
None

How can this be verified by QA?
By ensuring other functionality in the gdk still works as expected.

#### Documentation
changelog

#### Primary reviewers
@oblm
